samples/performance/async_compute/async_compute.cpp

```diff
@@ -0,0 +1,820 @@
+		{
+			vkb::sg::LightProperties props{};
+			props.color = glm::vec3(50.0f, 40.0f, 30.0f);
+			light->set_properties(props);
+			auto *node = light->get_node();
+
+			// Hardcoded to fit to the scene.
+			auto ortho_camera = std::make_unique<vkb::sg::OrthographicCamera>("shadow_camera",
+			                                                                  -2000, 3000,
+			                                                                  -2500, 1500,
+			                                                                  -2000, 2000);
+
+			ortho_camera->set_node(*node);
+			scene->add_component(std::move(ortho_camera), *node);
+			shadow_camera = &node->get_component<vkb::sg::Camera>();
+			break;
+		}
+	}
+
+	prepare_render_targets();
+
+	vkb::ShaderSource vert_shader("async_compute/forward.vert");
+	vkb::ShaderSource frag_shader("async_compute/forward.frag");
+	auto              scene_subpass = std::make_unique<ShadowMapForwardSubpass>(get_render_context(),
+                                                                   std::move(vert_shader), std::move(frag_shader),
+                                                                   *scene, *camera,
+                                                                   *shadow_camera);
+
+	vkb::ShaderSource shadow_vert_shader("async_compute/shadow.vert");
+	vkb::ShaderSource shadow_frag_shader("async_compute/shadow.frag");
+	auto              shadow_scene_subpass = std::make_unique<DepthMapSubpass>(get_render_context(),
+                                                                  std::move(shadow_vert_shader), std::move(shadow_frag_shader),
+                                                                  *scene, *shadow_camera);
+	shadow_render_pipeline.add_subpass(std::move(shadow_scene_subpass));
+
+	vkb::ShaderSource composite_vert_shader("async_compute/composite.vert");
+	vkb::ShaderSource composite_frag_shader("async_compute/composite.frag");
+	auto              composite_scene_subpass =
+	    std::make_unique<CompositeSubpass>(get_render_context(), std::move(composite_vert_shader), std::move(composite_frag_shader));
+
+	forward_render_pipeline.add_subpass(std::move(scene_subpass));
+	forward_render_pipeline.set_load_store({{VK_ATTACHMENT_LOAD_OP_CLEAR, VK_ATTACHMENT_STORE_OP_STORE},
+	                                        {VK_ATTACHMENT_LOAD_OP_CLEAR, VK_ATTACHMENT_STORE_OP_DONT_CARE}});
+
+	vkb::RenderPipeline blit_render_pipeline;
+	blit_render_pipeline.add_subpass(std::move(composite_scene_subpass));
+	blit_render_pipeline.set_load_store({{VK_ATTACHMENT_LOAD_OP_CLEAR, VK_ATTACHMENT_STORE_OP_STORE},
+	                                     {VK_ATTACHMENT_LOAD_OP_CLEAR, VK_ATTACHMENT_STORE_OP_DONT_CARE}});
+
+	set_render_pipeline(std::move(blit_render_pipeline));
+
+	stats->request_stats({
+	    vkb::StatIndex::frame_times,
+	    vkb::StatIndex::gpu_cycles,
+	    vkb::StatIndex::gpu_vertex_cycles,
+	    vkb::StatIndex::gpu_fragment_cycles,
+	});
```

Comment:
I think you need to add "vkb::CounterSamplingConfig{vkb::CounterSamplingMode::Continuous}" as a second argument to request_stats here. That will prevent the Vulkan extension for performance counters being used.

The problem with the Vulkan counters extension is that it is designed to operate on a single command buffer (it's a query pool thing). This sample uses at least four different command buffers, each of which is starting and stopping stats sampling, and the Vulkan counters then just don't work.

You can then remove all the begin_sampling() and end_sampling() calls.

Reply:
Ye, I guess I can fix that.

Reply:
Continous seems to be broken. The output is very jumpy and weird, and I don't know why. I removed begin/end_sampling which should work. I cannot reproduce any issues here. I just pushed a commit which removes the begin/end_sampling calls.

Reply:
The pipeline_barriers sample uses continuous sampling. Does that work correctly?
We don't support continuous sampling here, someone at Arm would need to look into that.

Reply:
> I removed begin/end_sampling which should work.

It does, but you'll also need to remove the stats that are listed. Right now, they appear in the UI but are always 0, since nothing is updating them any more.

Reply:
Found some bugs in the Continuous implementation. Fixed those, and flipped the sample over to continuous.

Reply:
Thanks a lot. That looks much better.